### PR TITLE
[Sim/Vis/GameState/SORRY] Vis feature metrics

### PR DESCRIPTION
**Pull request by Eirikalb**
# Summary

Provides some new metrics, added a div zero check to computeUnmitigatedDisasterEffects()
New Metrics:
- Rules broken
- Commonpool mitigation of disasters
- Number of disasters survived
- Impact of disasters survived
Also changed some colours for the foraging graph to make a clear distinction between fishing and deerhunting
image
image

## Additional Information

Auxiliary information goes here.

## Test Plan

- Add information on how you tested your changes.

### CODE REVIEW

File: internal/common/disasters/environmentdefinition.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ type disasterParameters struct {
 type DisasterReport struct {
 	Magnitude shared.Magnitude
 	X, Y      shared.Coordinate
+	Effects   DisasterEffects
 }
```
Comment (lhl2617) [inline internal/common/disasters/environmentdefinition.go:36]: Wondering if this is copied correctly as part of GameState?

Comment (lhl2617) [reply internal/common/disasters/environmentdefinition.go:36]: structs are reference types

File: website/src/components/Visualisations/Metrics/MetricEntries.ts [hunk 2]
```diff
@@ -122,6 +129,70 @@ const returnsFromCriticalMetricCollection = (data: OutputJSONType) => {
     return metrics
 }
 
+const rulesBrokenPerTeamCollection = (data: OutputJSONType) => {
+    const metrics = emptyMetrics()
+    // Since IIGOHistories is repeated, take the one from the LAST GameState and
+    // do Object.entries to make it iterable. List of array'ed tuples.
+    data.GameStates.forEach((gameState) => {
+        if (gameState.RulesBrokenByIslands !== null) {
+            Object.entries(gameState.RulesBrokenByIslands).forEach((island) => {
```
Comment (lhl2617) [inline website/src/components/Visualisations/Metrics/MetricEntries.ts:138]: Prefer Object.entries(obj).forEach([k, v]) => ...
with well named k and v

**Review by lhl2617 — APPROVED** [review]
ok if checked copy works correctly

File: internal/common/disasters/environmentdefinition.go (hunk 1 above)
Comment (Eirikalb) [reply internal/common/disasters/environmentdefinition.go:36]: It is in the gamestate but it seems that the first entry is null for the Effects fields...

Comment (Eirikalb) [reply internal/common/disasters/environmentdefinition.go:36]: The copy works correctly, but I the DisasterReport does not have a default value so the first round copies nulls.

**Pull request merged**